### PR TITLE
 Keyboard navigation in goals list

### LINKED ISSUE #914: Keyboard navigation in goals list
If any in focus and preview is open user may switch between goals via up and down arrows on keyboard. Space/Enter to open/close preview.

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Feature

## Related issues

Resolve #914

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/9020793/2921a2cb-d8b8-4b84-8522-083a19606dca

### CODE REVIEW

File: src/components/DashboardPage/DashboardPage.tsx [hunk 1]
```diff
@@ -132,45 +141,54 @@ export const DashboardPage = ({ user, ssrTime, defaultPresetFallback }: External
...
+                        <ProjectListItemCollapsable
+                            key={project.id}
+                            interactive={false}
+                            visible
+                            project={project}
+                            href={routes.project(project.id)}
+                            goals={goals.map((g) => (
+                                <TreeViewElement key={g.id}>
+                                    <ListViewItem
+                                        key={g.id}
```
Comment (awinogradov) [inline src/components/DashboardPage/DashboardPage.tsx:155]: U need key prop on root elem only

File: src/components/FlatGoalList.tsx [hunk 2]
```diff
@@ -78,39 +78,58 @@ export const FlatGoalList: React.FC<GoalListProps> = ({ queryState, setTagFilter
...
-                            achivedCriteriaWeight={g._achivedCriteriaWeight}
-                            onTagClick={setTagFilterOutside}
+                {goalsOnScreen?.map((g) => {
+                    return (
+                        <ListViewItem
+                            key={g.id}
+                            value={g}
+                            renderItem={({ active, hovered: _, ...props }) => (
+                                <TableRowItem
+                                    key={g.id}
```
Comment (awinogradov) [inline src/components/FlatGoalList.tsx:101]: Unnecessary key

File: src/components/GlobalSearch/GlobalSearch.tsx [hunk 3]
```diff
@@ -133,84 +133,87 @@ export const GlobalSearch = () => {
...
-                                                        },
-                                                        { name: 'projectId', columnProps: { col: 3 } },
-                                                        { name: 'issuers' },
-                                                        { name: 'estimate', columnProps: { width: '8ch' } },
-                                                    ]}
-                                                    {...props}
-                                                />
-                                            </NextLink>
+                                            <StyledProjectListItemCompact
+                                                key={item.id}
```
Comment (awinogradov) [inline src/components/GlobalSearch/GlobalSearch.tsx:202]: Unnecessary key

File: src/components/GroupedGoalList.tsx [hunk 4]
```diff
@@ -70,8 +70,17 @@ export const GroupedGoalList: React.FC<GroupedGoalListProps> = ({ queryState, se
         return pages.flatMap((page) => page.projects);
     }, [data]);
 
+    const handleItemEnter = useCallback(
+        (goal: GoalByIdReturnType) => {
+            if (goal) {
```
Comment (awinogradov) [inline src/components/GroupedGoalList.tsx:75]: Why this logic not in onKeyboardClick? Do we really need call callback if argument and current element are undefined?

File: src/components/ProjectListItemConnected.tsx [hunk 5]
```diff
@@ -84,31 +84,40 @@ export const ProjectListItemConnected: FC<ProjectListItemConnectedProps> = ({
...
-                                watching={g._isWatching}
-                                achivedCriteriaWeight={g._achivedCriteriaWeight}
-                                onTagClick={onTagClick}
-                            />
-                        </TableRowItem>
+                        <ListViewItem
+                            key={g.id}
+                            value={g}
+                            renderItem={({ active, hovered: _, ...props }) => (
+                                <div>
```
Comment (awinogradov) [inline src/components/ProjectListItemConnected.tsx:91]: ?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GlobalSearch/GlobalSearch.tsx (hunk 3 above)
Comment (asabotovich) [reply src/components/GlobalSearch/GlobalSearch.tsx]: fixed

File: src/components/FlatGoalList.tsx (hunk 2 above)
Comment (asabotovich) [reply src/components/FlatGoalList.tsx]: fixed

File: src/components/DashboardPage/DashboardPage.tsx (hunk 1 above)
Comment (asabotovich) [reply src/components/DashboardPage/DashboardPage.tsx]: fixed

File: src/components/ProjectListItemConnected.tsx (hunk 5 above)
Comment (asabotovich) [reply src/components/ProjectListItemConnected.tsx]: fixed

File: src/components/GroupedGoalList.tsx (hunk 4 above)
Comment (asabotovich) [reply src/components/GroupedGoalList.tsx:75]: fixed

File: src/components/DashboardPage/DashboardPage.tsx [hunk 6]
```diff
@@ -121,6 +121,15 @@ export const DashboardPage = ({ user, ssrTime, defaultPresetFallback }: External
             ? currentPreset.description
             : tr('This is your personal goals bundle');
 
+    const handleItemEnter = useCallback(
+        (goal: GoalByIdReturnType) => {
+            if (goal) {
```
Comment (awinogradov) [inline src/components/DashboardPage/DashboardPage.tsx:126]: Not fixed

File: src/components/GroupedGoalList.tsx (hunk 4 above)
Comment (awinogradov) [reply src/components/GroupedGoalList.tsx:75]: nope

File: src/components/ProjectPage/ProjectPage.tsx [hunk 7]
```diff
@@ -125,6 +125,15 @@ export const ProjectPage = ({ user, ssrTime, params: { id }, defaultPresetFallba
 
     const { toggleProjectWatching, toggleProjectStar } = useProjectResource(id);
 
+    const handleItemEnter = useCallback(
+        (goal: GoalByIdReturnType) => {
+            if (goal) {
```
Comment (awinogradov) [inline src/components/ProjectPage/ProjectPage.tsx:130]: Not fixed

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Comment by asabotovich** [comment]
https://github.com/taskany-inc/issues/assets/9020793/b1176821-9dbc-4910-b985-de1cec81e503

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**